tidb_binlog/node/node.go

```diff
@@ -1,21 +1,51 @@
 package node
 
 import (
-	pb "github.com/pingcap/tipb/go-binlog"
+	"time"
 )
 
-// Status describes the status information of a tidb-binlog node in etcd
+// State is the state of node.
+type State string
+
+const (
+	// Online means the node can receive request.
+	Online State = "online"
+
+	// Pausing means the node is pausing.
+	Pausing State = "pausing"
+
+	// Paused means the node is already paused.
+	Paused State = "paused"
+
+	// Closing means the node is closing, and the state will be Offline when closed.
+	Closing State = "closing"
+
+	// Offline means the node is offline, and will not provide service.
+	Offline State = "offline"
+
+	// Unknow means the node's state is unkonw.
+	Unknow State = "unknow"
+)
+
+// Status describes the status information of a tidb-binlog node in etcd.
 type Status struct {
+	// the id of node.
 	NodeID string
-	Host   string
-	// send hearbeats to keep alive
-	IsAlive bool
-	// denote whether node is offline
-	IsOffline bool
-	//  position of latest binlog that write into binlog file of pump
-	LatestFilePos pb.Pos
-	// offset of latest binlog that write into kafka by pump
-	LatestKafkaPos pb.Pos
-	//  offline TSO
-	OfflineTS int64
+
+	// the host of pump or node.
+	Host string
+
+	// the state of pump.
+	State State
+
+	// the score of node, it is report by node, calculated by node's qps, disk usage and binlog's data size.
+	// if Score is less than 0, means this node is useless. Now only used for pump.
+	Score int64
```

Comment:
set by pump? i think better calculated by client

Reply:
if we need update the method to calculate score, we just need update pump. But if we calculate it in pump client, we need update tidb.